src/Forms/DatetimeField.php

```diff
@@ -58,59 +85,204 @@ class DatetimeField extends FormField
      */
     protected $dateTimeOrder = '{date} {time}';
 
-    public function __construct($name, $title = null, $value = "")
-    {
-        $this->timeField = TimeField::create($name . '[time]', false);
-        $this->dateField = DateField::create($name . '[date]', false);
-        parent::__construct($name, $title, $value);
-    }
+    /**
+     * Custom timezone
+     *
+     * @var string
+     */
+    protected $timezone = null;
 
     public function setForm($form)
     {
         parent::setForm($form);
-        $this->dateField->setForm($form);
-        $this->timeField->setForm($form);
         return $this;
     }
 
     public function setName($name)
     {
         parent::setName($name);
-        $this->dateField->setName($name . '[date]');
-        $this->timeField->setName($name . '[time]');
+        return $this;
+    }
+
+    public function getAttributes()
+    {
+        $attributes = parent::getAttributes();
+
+        $attributes['lang'] = i18n::convert_rfc1766($this->getLocale());
+
+        if ($this->getHTML5()) {
+            $attributes['type'] = 'datetime-local';
+            $attributes['min'] = $this->getMinDatetime();
+            $attributes['max'] = $this->getMaxDatetime();
+        }
+
+        return $attributes;
+    }
+
+    public function getSchemaDataDefaults()
+    {
+        $defaults = parent::getSchemaDataDefaults();
```

Comment:
We should add the min / max here. Please add these (and move the `html5` key) to the `data` property, rather than a top level key; `data` is intended for field-specific values, and top level keys are common across all fields.